src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionAnnotatedTypeFactory.java

```diff
@@ -194,6 +201,62 @@ public DependencyInjectionAnnotatedTypeFactory(BaseTypeChecker c) {
     this.postInit();
   }
 
+  /**
+   * Returns the string representation of the given type mirror. If the type mirror is a primitive
+   * or boxed type, the string representation is the type kind. Otherwise, the string representation
+   * is the fully-qualified name of the type.
+   *
+   * @param elementTypeMirror the type mirror
+   * @return the string representation of the given type mirror
+   */
+  protected static String resolveInjectionPointClassName(TypeMirror elementTypeMirror) {
+    TypeKind injectionKind = TypeKindUtils.primitiveOrBoxedToTypeKind(elementTypeMirror);
+    return injectionKind != null ? injectionKind.toString() : elementTypeMirror.toString();
+  }
+
+  /**
+   * Returns the resolved name of the given injection point class name. If the given injection point
+   * class name is a primitive or boxed type, the resolved name is the type kind. Otherwise, the
+   * resolved name is the given injection point class name.
+   *
+   * @param injectionClassName the fully-qualified class name of the injection point
+   * @return the resolved name of the given injection point class name
+   */
+  protected static String resolveInjectionPointClassName(String injectionClassName) {
+    String resolvedName;
```

Comment:
Rather than using this variable and `break` statements, have the switch cases `return` the correct value directly.